src/query/src/promql/planner.rs

```diff
@@ -1966,27 +1982,81 @@ impl PromPlanner {
             .field_columns
             .iter()
             .map(|col| {
-                DfExpr::AggregateFunction(AggregateFunction {
+                Ok(DfExpr::AggregateFunction(AggregateFunction {
                     func: aggr.clone(),
                     args: vec![DfExpr::Column(Column::from_name(col))],
                     distinct: false,
                     filter: None,
                     order_by: None,
                     null_treatment: None,
-                })
+                }))
             })
-            .collect();
+            .collect::<Result<Vec<_>>>()?;
+
+        // if the aggregator is `count_values`, it must be grouped by current fields.
+        let prev_field_exprs = if op.id() == token::T_COUNT_VALUES {
+            let prev_field_exprs: Vec<_> = self
+                .ctx
+                .field_columns
+                .iter()
+                .map(|col| DfExpr::Column(Column::from_name(col)))
+                .collect();
 
-        // update value column name according to the aggregators
+            ensure!(
+                self.ctx.field_columns.len() == 1,
+                UnsupportedExprSnafu {
+                    name: "count_values on multi-value input"
+                }
+            );
+
+            prev_field_exprs
+        } else {
+            vec![]
+        };
+
+        // update value column name according to the aggregators,
         let mut new_field_columns = Vec::with_capacity(self.ctx.field_columns.len());
+
         let normalized_exprs =
             normalize_cols(exprs.iter().cloned(), input_plan).context(DataFusionPlanningSnafu)?;
         for expr in normalized_exprs {
             new_field_columns.push(expr.schema_name().to_string());
         }
         self.ctx.field_columns = new_field_columns;
 
-        Ok(exprs)
+        Ok((exprs, prev_field_exprs))
```

Comment:
add some documents for this new return value

Reply:
Ok